src/jit/compiler.h

```diff
@@ -297,6 +297,208 @@ enum RefCountState
+//  the "liveRangeList" to set the barrier to nullptr or the last
+//  "VariableLiveRange" if it is opened.
+//  If no "VariableLiveRange" was edited/added during block,
+//  the iterator points to the end of variable's LiveRangeList.
+//
+class LiveRangeDumper
+{
+private:
+    LiveRangeListIterator m_StartingLiveRange; // Iterator to the first edited/added position
+                                               // during actual block code generation. If last
+                                               // block had a closed "VariableLiveRange" (with
+                                               // a valid "m_EndEmitLocation") and not changes
+                                               // were applied to variable liveness, it points
+                                               // to the end of variable's LiveRangeList.
+    bool m_hasLiveRangestoDump;                // True if a live range for this variable has been
+                                               // reported from last call to EndBlock
+
+public:
+    LiveRangeDumper(const LiveRangeList* liveRanges);
+
+    ~LiveRangeDumper(){};
+
+    // Make the dumper point to the last "VariableLiveRange" opened or nullptr if all are closed
+    void resetDumper(const LiveRangeList* list);
+
+    //  Make "LiveRangeDumper" instance points the last "VariableLifeRange" added so we can
+    // starts dumping from there after the actual "BasicBlock"s code is generated.
+    void setDumperStartAt(const LiveRangeListIterator liveRangeIt);
+
+    // Return an iterator to the first "VariableLiveRange" edited/added during the current
+    // "BasicBlock"
+    LiveRangeListIterator getStartForDump() const;
+
+    // Return whether at least a "VariableLiveRange" was alive during the current "BasicBlock"'s
+    // code generation
+    bool hasLiveRangesToDump() const;
+};
+#endif // DEBUG
+
+//--------------------------------------------
+//
+// VariableLiveDescriptor: This class persist and update all the changes
+//  on the variable home of a variable. It has an instance of "LiveRangeList"
+//  and methods to report the start/end of a VariableLiveRange.
+//
+// Notes:
+//  It has an instance of "LiveRangeList"
+//  and methods to report the start/end of a VariableLiveRange.
+//
+class VariableLiveDescriptor
+{
+private:
+    LiveRangeList* m_VariableLiveRanges; // the variable homes of this variable
+
+#ifdef DEBUG
+    LiveRangeDumper* m_VariableLifeBarrier;
+#endif // DEBUG
+
+public:
+    VariableLiveDescriptor(CompAllocator allocator);
+
+    ~VariableLiveDescriptor();
+
+    bool hasVariableLiveRangeOpen() const;
+
+    size_t getAmountLiveRanges() const;
+
+    const LiveRangeList* getLiveRanges() const;
+
+    void startLiveRangeFromEmitter(CodeGenInterface::siVarLoc varHome, emitter* _emitter) const;
+
+    void endLiveRangeAtEmitter(emitter* _emitter) const;
+
+    void updateLiveRangeAtEmitter(CodeGenInterface::siVarLoc varHome, emitter* _emitter) const;
```

Comment:
I find it a bit odd that functions like start/end/update are `const`. That's a consequence of that list being dynamically allocated.

Reply:
I agree with you, its not logical const-ness. I can remove them.

Reply:
I am still thinking about this. Is it better to have to have physical const-ness than nothing?

Reply:
I'd say that as it is now it's misleading, the logical state of the object does change. But then what exactly `const` means for an object is up to the class so I suppose this is a bit philosophical.